fuse_examples/classification/multimodality/model_tabular_imaging.py

```diff
@@ -0,0 +1,295 @@
+            imaging_features = F.max_pool2d(imaging_features, kernel_size=imaging_features.shape[2:])
+
+        elif self.pooling == 'avg':
+            imaging_features = F.avg_pool2d(imaging_features, kernel_size=imaging_features.shape[2:])
+
+        imaging_features = self.projection_imaging.forward(imaging_features)
+        imaging_features = torch.squeeze(torch.squeeze(imaging_features,dim=3),dim=2)
+        return imaging_features
+
+
+    def forward(self, batch_dict):
+
+        imaging_features = FuseUtilsHierarchicalDict.get(batch_dict, 'model.imaging_features')
+        tabular_features = FuseUtilsHierarchicalDict.get(batch_dict, 'model.tabular_features')
+        imaging_features = self.fix_imaging(imaging_features)
+        res = torch.cat([tabular_features, imaging_features], dim=1)
+        return res
+
+
+#Tabular model
+class FuseModelTabularContinuousCategorical(torch.nn.Module):
+    def __init__(self,
+                 continuous_tabular_input: Tuple[Tuple[str, int], ...],
+                 categorical_tabular_input: Tuple[Tuple[str, int], ...],
+                 backbone_categorical_tabular: FuseMultilayerPerceptronBackbone,
+                 backbone_continuous_tabular: FuseMultilayerPerceptronBackbone,
+                 heads: Sequence[torch.nn.Module],
+                 ) -> None:
+        super().__init__()
+        self.continuous_tabular_input = continuous_tabular_input
+        self.categorical_tabular_input = categorical_tabular_input
+        self.backbone_categorical_tabular = backbone_categorical_tabular
+        self.backbone_cat_tabular = backbone_continuous_tabular
+        # self.add_module('backbone', self.backbone)
+        self.heads = torch.nn.ModuleList(heads)
+        self.add_module('heads', self.heads)
+
+    def forward(self, batch_dict: Dict) -> Dict:
+        categorical_input = FuseUtilsHierarchicalDict.get(batch_dict, self.categorical_tabular_input[0][0])
+        categorical_embeddings = self.backbone_categorical_tabular(categorical_input)
+        continuous_input = FuseUtilsHierarchicalDict.get(batch_dict, self.continuous_tabular_input[0][0])
+        input_cat = torch.cat((categorical_embeddings, continuous_input), 1)
+        tabular_features = self.backbone_cat_tabular(input_cat)
+        FuseUtilsHierarchicalDict.set(batch_dict, 'model.tabular_features', tabular_features)
+
+        for head in self.heads:
+            batch_dict = head.forward(batch_dict)
+        return batch_dict['model']
+
+#Tabular Imaging model
+class FuseMultiModalityModel(torch.nn.Module):
+    def __init__(self,
+                 tabular_inputs: Tuple[Tuple[str, int], ...]=None,
+                 imaging_inputs: Tuple[Tuple[str, int], ...]=None,
+                 tabular_backbone: torch.nn.Module=None,
+                 imaging_backbone: torch.nn.Module=None,
+                 multimodal_backbone: torch.nn.Module=None,
+                 tabular_heads: Sequence[torch.nn.Module]=None,
```

Comment:
Why do you care if it's imaging, tabular or multimodal heads?
If you don't care, you can maintain a single list for all heads.